### PR TITLE
Solution-wide config support for `EnsureSchemaIsUpToDate`

### PR DESCRIPTION
**Pull request by martinothamar**
#### Description

Right now, having a solution level config file without a project level one is supported. But `EnsureSchemaIsUpToDate` doesn't care about that and will therefore throw an exception if it doesn't find a config file in the project directory (it only checks for project level, even though earlier in the callstack if neither project or solution level one is found a project level one is generated automatically).
This PR avoids throwing exception during `EnsureSchemaIsUpToDate` when a solution config is specified.

#### The solution

During `EnsureSchemaIsUpToDate`, a `string solutionDirectory` argument has been added, so that both project and solution file can be checked. Will only try to ensure schema is up to date if the respective file exists.

#### Todos

 * [ ] Related issues
 * [x] Tests
 * [ ] Documentation

### CODE REVIEW

File: Fody/ConfigFileFinder/ConfigFileFinder.cs [hunk 1]
```diff
@@ -99,7 +99,8 @@ public static WeaverConfigFile GenerateDefault(string projectDirectory, List<Wea
 
     static void CreateSchemaForConfig(string projectConfigFilePath, IEnumerable<WeaverEntry> weavers)
     {
-        var schema = XDocument.Parse(Fody.Properties.Resources.FodyWeavers_SchemaTemplate);
+        var template = Fody.Properties.Resources.FodyWeavers_SchemaTemplate;
```
Comment (tom-englert) [inline Fody/ConfigFileFinder/ConfigFileFinder.cs]: Do not introduce unnecessary changes

**Comment by SimonCropp** [comment]
sorry for the delay on this one. i will smoke test it and deploy now
